src/main/java/org/openrewrite/java/spring/boot2/IntegrationSchedulerPoolRecipe.java

```diff
@@ -0,0 +1,135 @@
+package org.openrewrite.java.spring.boot2;
+
+import org.openrewrite.*;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.marker.Markers;
+import org.openrewrite.maven.MavenVisitor;
+import org.openrewrite.maven.tree.Maven;
+import org.openrewrite.maven.tree.Pom;
+import org.openrewrite.maven.tree.Scope;
+import org.openrewrite.properties.PropertiesVisitor;
+import org.openrewrite.properties.tree.Properties;
+
+import java.nio.file.Path;
+import java.util.*;
+
+/**
+ * @author Alex Boyko
+ */
+public class IntegrationSchedulerPoolRecipe extends Recipe {
+
+    private static final String SCHEDULER_PROPERTY_ENTRY = "SchedulerPropertyIndex";
+
+    private static final String MIGRATION_MESSAGE = " TODO: Consider Scheduler thread pool size for Spring Integration";
+
+    @Override
+    public String getDisplayName() {
+        return "Integration Sceduler Pool Size";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Spring Integration now reuses an available TaskScheduler rather than configuring its own. In a" +
+                " typical application setup relying on the auto-configuration, this means that Spring Integration" +
+                " uses the auto-configured task scheduler that has a pool size of 1. To restore Spring Integration’s" +
+                " default of 10 threads, use the spring.task.scheduling.pool.size property.";
+    }
+
+//    @Override
+//    protected @Nullable TreeVisitor<?, ExecutionContext> getApplicableTest() {
+//        return new MavenVisitor() {
+//            @Override
+//            public Maven visitMaven(Maven maven, ExecutionContext ctx) {
+//                Collection<Pom.Dependency> deps = maven.getModel().getDependencies(Scope.Compile);
+//                boolean boot25 = false;
+//                boolean si = false;
+//                for (Pom.Dependency d : maven.getModel().getDependencies(Scope.Compile)) {
+//                    if (!boot25) {
+//                        boot25 = d.getArtifactId().equals("spring-boot")
+//                                && d.getVersion().startsWith("2.5.")
+//                                && d.getGroupId().equals("org.springframework.boot");
+//                    }
+//                    if (!si) {
+//                        si = d.getArtifactId().equals("spring-integration-core")
+//                                && d.getGroupId().equals("org.springframework.integration");
+//                    }
+//                    if (boot25 && si) {
+//                        break;
+//                    }
+//                }
+//                if (boot25 && si) {
+//                    return maven.withMarkers(maven.getMarkers().searchResult());
+//                }
+//                return maven;
+//            }
+//        };
+//    }
+
+    @Override
+    protected List<SourceFile> visit(List<SourceFile> before, ExecutionContext ctx) {
```

Comment:
Hi @BoykoAlex, just curious if you have tried leveraging your `applicableTest`, returning a `PropertiesVisitor` from `Recipe.getVisitor` override, and adding a singleSourceApplicable test to check for the `application.properties' file?

Reply:
I was also thinking of running a YamlVisitor to modify application yaml file. I could specify both yaml and properties file filters in singleSourceApplicable test but how shall i proceed with `getVisitor()` then?

Reply:
one idea is to convert the yaml and properties visitors to recipes which can scheduled to run next from the applicableTest
```java
if (boot25 && si) {
     doAfterVisit(new IntegrationSchedulerPoolYamlRecipe());
     doAfterVisit(new IntegrationSchedulerPoolPropertiesRecipe());
```

Reply:
although, maybe it's just better to use the before method instead